src/test/java/org/jitsi/meet/test/MuteTest.java

```diff
@@ -247,13 +247,17 @@ private void muteParticipant1BeforeParticipant2JoinsAndScreenshare(String enable
 
         WebParticipant participant1 = getParticipant1();
         WebParticipant participant2 = joinSecondParticipant(url2);
-        participant2.waitToJoinMUC();
-        participant2.waitForIceConnected();
-        participant2.waitForSendReceiveData(true, false);
 
-        // Wait for the media to switch over to the p2p connection.
         TestUtils.waitMillis(2000);
 
+        // Wait for the media to switch over to the p2p connection in the p2p test.
+        if (enableP2p.equals("true")) {
+            MeetUtils.waitForP2PIceConnected(participant1.getDriver());
```

Comment:
Did you want `participant2.getDriver()`  here? 

nit: The brace should be on a new line.

Reply:
Fixed the brace. It shouldn't matter if we check for ICE connection establishment on p1 or p2 right ?

Reply:
Well...I'm not sure, it's confusing.

If we can safely call `waitForSendReceiveData` before ICE has completed, why do we need to separately wait for ICE? Otherwise, it does make a difference if we check p1 or p2.

Reply:
In the P2P case, we are waiting for the P2P connection to be established. That would mean that the media has switched over to the p2p connection and then only we proceed with the real test.

Reply:
The p2p media session on both p1 and p2 gets created only when p2 joins the call so checking for the p2p connection state on either of them should work, right ?